holoviews/ipython/__init__.py

```diff
@@ -140,8 +139,21 @@ class notebook_extension(param.ParameterizedFunction):
 
     _loaded = False
 
+    _backends = {'matplotlib': 'mpl',
+                 'bokeh': 'bokeh'}
+
     def __call__(self, *args, **params):
-        resources = self._get_resources(args, params)
+        imports = [(name, b) for name, b in self._backends.items()
+                   if name in args or params.get(name, False)]
+        if not imports:
+            imports.append(('matplotlib', 'mpl'))
+        for backend, imp in imports:
+            try:
+                __import__('.'.join(['holoviews.plotting', imp]))
```

Comment:
Don't see why you would use join instead of a simple formatting string in this instance.